code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -540,3 +544,207 @@
 	REMOVE_TRAIT(owner, TRAIT_NODEATH, type)
 	owner.remove_movespeed_modifier(/datum/movespeed_modifier/reagent/nooartrium)
 	owner.remove_actionspeed_modifier(/datum/actionspeed_modifier/nooartrium)
+
+/*				Non c2 medicines 				*/
+
+/datum/reagent/impurity/mannitol
+	name = "Mannitoil"
+	description = "Gives the patient a temporary speech impediment."
+	color = "#CDCDFF"
+	addiction_types = list(/datum/addiction/medicine = 5)
+	ph = 12.4
+	liver_damage = 0
+	///The speech we're forcing on the owner
+	var/speech_option
+
+/datum/reagent/impurity/mannitol/on_mob_add(mob/living/owner, amount)
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon = owner
+	speech_option = pick(list(SWEDISH, UNINTELLIGIBLE, STONER, MEDIEVAL, WACKY, NERVOUS, MUT_MUTE))
+	carbon.dna?.activate_mutation(speech_option)
+
+/datum/reagent/impurity/mannitol/on_mob_delete(mob/living/owner)
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon = owner
+	carbon.dna?.remove_mutation(speech_option)
+
+/datum/reagent/inverse/neurine
+	name = "Neruwhine"
+	description = "Induces a temporary brain trauma in the patient by redirecting neuron activity."
+	color = "#DCDCAA"
+	ph = 13.4
+	addiction_types = list(/datum/addiction/medicine = 8)
+	metabolization_rate = 0.025 * REM
+	tox_damage = 0
+	//The temporary trauma passed to the owner
+	var/datum/brain_trauma/temp_trauma
+
+/datum/reagent/inverse/neurine/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	.=..()
+	if(temp_trauma)
+		return
+	if(!(DT_PROB(creation_purity*10, delta_time)))
+		return
+	var/traumalist = subtypesof(/datum/brain_trauma)
+	traumalist -= /datum/brain_trauma/severe/split_personality //Uses a ghost, I don't want to use a ghost for a temp thing.
+	var/datum/brain_trauma/trauma = pick(traumalist)
+	var/obj/item/organ/brain/brain = owner.getorganslot(ORGAN_SLOT_BRAIN)
+	if(brain.brain_gain_trauma(trauma, TRAUMA_RESILIENCE_MAGIC))
+		temp_trauma = trauma
```

Comment:
isn't there some kind of proc that already exists for this?

also, what happens if this tries to give the patient a trauma that they already have?

Reply:
Is there? I can't seem to find it. At least it's not under the brain_item obj calls.

I'll change the latter to check